core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/AgoMatcher.java

```diff
@@ -0,0 +1,22 @@
+package uk.ac.bbsrc.tgac.miso.core.util;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class AgoMatcher {
+  private final Pattern pattern;
+  private final int seconds;
+
+  AgoMatcher(String unit, int seconds) {
+    pattern = Pattern.compile("^([0-9]+)" + unit + "s?$", Pattern.CASE_INSENSITIVE);
+    this.seconds = seconds;
+  }
+
+  public int matches(String text) {
```

Comment:
Please rename to something that conveys that this does more than check for a match. Preferably Javadoc to make it more clear (also gets number of seconds, which can't be 0 because that means no match)